crypto_kx/src/keys/public.rs

```diff
@@ -0,0 +1,48 @@
+use core::convert::TryFrom;
+
+use crate::errors::InvalidLength;
+
+/// [`Key`] which can be freely shared.
+#[derive(Clone, Copy, Hash, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(our_serde::Deserialize, our_serde::Serialize)
+)]
+#[cfg_attr(feature = "serde", serde(crate = "our_serde"))]
+pub struct Key(x25519_dalek::PublicKey);
```

Comment:
It would probably be good to name this `PublicKey` as it's not clear from the name otherwise whether it's public or secret